api/agent/slots.go

```diff
@@ -131,19 +122,13 @@ func (a *slotQueue) startDequeuer(ctx context.Context) chan *slotToken {
 				return
 			}
 
-			// pop
 			item := a.slots[len(a.slots)-1]
-			a.slots = a.slots[:len(a.slots)-1]
```

Comment:
i'm finding it challenging to audit the correctness of the process of acquiring a slot now. it now seems like we are relying on acquireSlot to do this, however after the loop in runHotReq we appear to do a slot.Close if we acquire the slot, which now appears to be something we have to do regardless of whether we are trying to eject -- so I'm confused how hot functions are running at all since we seem to be closing every slot we acquire?

Reply:
in runHot(), if acquire succeeds, then it means consumer did not acquire it, then we close and exit processing (since it's timeout, shutdown, etc.)

on the consumer side, if acquire fails, it means the token was ejected and consumer must get another one.

Reply:
that's right. so we just moved slot close up to the caller, that makes sense